File: Graph/WeirdAlgorithm.java
import java.util.Scanner;

public class WeirdAlgorithm {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Read the input number
        long n = sc.nextLong(); // Using long to handle larger values

        // Print the initial number
        System.out.print(n + " ");

        // Loop until n becomes 1
        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2; // If n is even, divide by 2
            } else {
                n = n * 3 + 1; // If n is odd, multiply by 3 and add 1
            }

            // Print the current value of n
            System.out.print(n + " ");
        }

        sc.close();
    }
}

File: CSES/Introduction_Problems/weird_algorithm.cpp
#include <iostream>

using namespace std;

int main() {
    // We use long long because the values in the sequence 
    // can exceed the range of a 32-bit integer (2^31 - 1).
    long long n;
    
    // Read the input value
    if (!(cin >> n)) return 0;

    // Print the starting value
    cout << n;

    // Continue the loop until n reaches 1
    while (n != 1) {
        if (n % 2 == 0) {
            // If even, divide by 2
            n /= 2;
        } else {
            // If odd, multiply by 3 and add 1
            n = 3 * n + 1;
        }
        // Print the next value in the sequence with a space
        cout << " " << n;
    }

    cout << endl;
    return 0;
}